apps/examples/testcase/le_tc/drivers/tc_dev_zero.c

```diff
@@ -0,0 +1,77 @@
+/****************************************************************************
+ *
+ * Copyright 2017 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/// @file tc_dev_zero.c
+/// @brief Test Case Example for dev_zero driver
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <fcntl.h>
+#include "tc_internal.h"
+#ifndef CONFIG_DISABLE_POLL
+#include <poll.h>
+#endif
+
+/****************************************************************************
+ * Name: dev_zero driver test
+ ****************************************************************************/
+/**
+* @fn                   :dev_zero_main
+* @brief                :Test the dev_zero driver. Since the driver is extremely small, all tests are done here
+* @scenario             :Test the dev_zero driver
+* API's covered         :open, read, write, poll
+* Preconditions         :none
+* Postconditions        :none
+* @return               :void
+*/
+
+void dev_zero_main(void)
+{
+	int fd = 0;
+	int ret = 0;
+	char *buf = malloc(100);
+
+	fd = open("/dev/zero", O_RDWR);
+	TC_ASSERT_GT("dev_zero_tc open", fd, 0);
```

Comment:
if open fail, buf cannot be freed.

Reply:
Done

Reply:
please check again. there is no free for buf when open is failed.

Reply:
Sorry. I had missed it out. 
Fixed it now.